Sources/DeepLearning/Layer.swift

```diff
@@ -268,3 +270,94 @@ public struct AvgPool2D<Scalar>: Layer
           kernelSize: poolSize, strides: strides, padding: padding)
     }
 }
+
+@_fixed_layout
+public struct LayerNorm<Scalar>: Layer
+    where Scalar : BinaryFloatingPoint & Differentiable & TensorFlowScalar {
+    /// The offset value, also known as beta.
+    public var offset: Tensor<Scalar>
+
+    /// The scale value, also known as gamma.
+    public var scale: Tensor<Scalar>
+
+    @noDerivative public let axis: Int32
+
+    /// The variance epsilon value.
+    @noDerivative public let epsilon: Tensor<Scalar>
+
+    public init(featureCount: Int,
+                axis: Int,
+                epsilon: Tensor<Scalar> = Tensor(0.001)) {
+        self.scale = Tensor<Scalar>(ones: [Int32(featureCount)])
+        self.offset = Tensor<Scalar>(zeros: [Int32(featureCount)])
+        self.axis = Int32(axis)
+        self.epsilon = epsilon
+    }
+
+    @differentiable(wrt: (self, input))
+    public func applied(to input: Tensor<Scalar>) -> Tensor<Scalar> {
+        let mean = input.mean(alongAxes: axis)
+        let variance = input.variance(alongAxes: axis)
+        let inv = rsqrt(variance + epsilon) * scale
+        return (input - mean) * inv + offset
+    }
+}
+
+public extension Tensor where Scalar : BinaryFloatingPoint,
+                       Scalar.RawSignificand : FixedWidthInteger {
+    @differentiable(wrt: (self) where Scalar : Differentiable)
+    func droppedOut(withProbability dropProbability: Double) -> Tensor {
+        let noise = Tensor(randomUniform: shape)
+        let keepMask = noise .>= Scalar(dropProbability)
+        let keepProbability = Scalar(1.0 - dropProbability)
+        return self * Tensor(keepMask) / Tensor(keepProbability)
+    }
+}
+
+@_fixed_layout
+public struct Dropout<Scalar>: Layer
+    where Scalar: BinaryFloatingPoint & Differentiable & TensorFlowScalar,
+          Scalar.RawSignificand : FixedWidthInteger {
+    @noDerivative public let dropProb: Double
+    @noDerivative public let mode: ModeRef
+    var _unused: Tensor<Scalar>
+
+    public init(dropProbability: Double, modeRef: ModeRef) {
```

Comment:
Just "probability" would be better.

```suggestion
    public init(probability: Double, modeRef: ModeRef) {
```

Reply:
Unfortunately, libraries haven’t standardized completely on whether the relevant scalar represents the “drop probability” or the “keep probability” so I’m not sure “probability” alone is necessarily enough

Reply:
My sense is that libraries that do mean "keep probability" have the word "keep" in the variable, e.g. `keep_prob`. Those meaning "drop probability" use "rate", "dropout rate" (which repeats the type name), or just "p". So I think the word "probability" in this type clearly indicates "dropout probability" without ambiguity.